Realm/RLMObjectSchema.mm

```diff
@@ -213,7 +213,7 @@ + (NSArray *)propertiesForClass:(Class)objectClass isSwift:(bool)isSwiftClass {
                 property.optional = true;
             }
             if (auto type = RLMCoerceToNil(propertyType)) {
-                if (existing == NSNotFound) {
+                if (existing == NSNotFound && ![ignoredProperties containsObject:propertyName]) {
```

Comment:
It makes sense. I moved the check to the first of the block https://github.com/realm/realm-cocoa/commit/036cb4abee37d75de536f35e8c38a7c15f439485. Is this what you mean?